python/ray/dag/compiled_dag_node.py

```diff
@@ -127,14 +127,15 @@ def _exec_task(self, task: "ExecutableTask", idx: int) -> bool:
     res = None
     try:
         res = input_reader.begin_read()
-    except ValueError as exc:
-        # ValueError is raised if a type hint was set and the returned
-        # type did not match the hint.
+    except IOError:
```

Comment:
QQ: should we also string match the error message, or is this certain IOError == channel closed in this case? (it could be wrong if we have different close API?)

Reply:
Hmm true, we should probably introduce a different Ray system error instead of using IOError. I'll add an issue to track this.